pkg/liqoctl/status/remoteinfo.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	controllerRuntimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/liqotech/liqo/pkg/utils/getters"
+)
+
+// RemoteInfoChecker implements the Check interface.
+// holds informations about remote clusters.
+type RemoteInfoChecker struct {
+	client             controllerRuntimeClient.Client
+	namespace          string
+	clusterNameFilter  *[]string
+	clusterIDFilter    *[]string
+	errors             bool
+	rootRemoteInfoNode InfoNode
+	collectionErrors   []collectionError
+}
+
+// newRemoteInfoChecker return a new remote info checker.
+func newRemoteInfoChecker(namespace string, clusterNameFilter, clusterIDFilter *[]string, client controllerRuntimeClient.Client) *RemoteInfoChecker {
+	return &RemoteInfoChecker{
+		client:             client,
+		namespace:          namespace,
+		clusterNameFilter:  clusterNameFilter,
+		clusterIDFilter:    clusterIDFilter,
+		errors:             false,
+		rootRemoteInfoNode: newRootInfoNode("Remote Clusters Information"),
+	}
+}
+
+// argsFilterCheck returns true if the given foreignClusterID or foreignClusterName are contained
+// in the respective filter list or if filter lists are void.
+func (ric *RemoteInfoChecker) argsFilterCheck(foreignClusterID, foreignClusterName string) bool {
+	return stringArrayContainsString(ric.clusterIDFilter, foreignClusterID) ||
+		stringArrayContainsString(ric.clusterNameFilter, foreignClusterName) ||
+		(len(*ric.clusterIDFilter) == 0 && len(*ric.clusterNameFilter) == 0)
+}
+
+// stringArrayContainsString returns true if the given string is contained in the given array.
+func stringArrayContainsString(a *[]string, s string) bool {
+	for _, v := range *a {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Use ContainsString (`utils/slice/slice.go`). In general, when a function looks common, please double check whether it already exists, otherwise extract it in an appropriate package, to attempt limit code duplication as much as possible.